### PR TITLE
dnsdist: Disable TLS renegotiation by default

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by Habbie — APPROVED** [review]

File: pdns/libssl.cc [hunk 1]
```diff
@@ -678,6 +678,12 @@ std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)> libssl_init_server_context(const TLS
 #endif /* SSL_OP_PRIORITIZE_CHACHA */
   }
 
+  if (!config.d_enableRenegotiation) {
+#ifdef SSL_OP_NO_RENEGOTIATION
+    sslOptions |= SSL_OP_NO_RENEGOTIATION;
+#endif
```
Comment (omoerbeek) [inline pdns/libssl.cc:684]: older openssl does not have `SSL_OP_NO_RENEGOTIATION` but does have `SSL_OP_NO_CLIENT_RENEGOTIATION`

Comment (omoerbeek) [reply pdns/libssl.cc:684]: > older openssl does not have `SSL_OP_NO_RENEGOTIATION` but does have `SSL_OP_NO_CLIENT_RENEGOTIATION`

correction, "older openssl" should be libressl...

Comment (rgacogne) [reply pdns/libssl.cc:686]: Pushed an untested fix to support `SSL_OP_NO_CLIENT_RENEGOTIATION` when available.

Comment (omoerbeek) [reply pdns/libssl.cc:686]: It looks good, but I wil test it tomorrow

**Review by omoerbeek — APPROVED** [review]
Confirmed the right `#ifdef` branch is taken on OpenBSD.

**Pull request merged**